src/features/journeys/store.ts

```diff
@@ -81,6 +81,20 @@ const journeysSlice = createSlice({
         remoteItem(journeyInstance.id, { data: journeyInstance })
       );
     },
+    journeyInstanceDeleted: (state, action: PayloadAction<number>) => {
+      const id = action.payload;
+
+      state.journeyInstanceList.items = state.journeyInstanceList.items.filter(
+        (item) => item.id !== id
+      );
```

Comment:
This object (`state`) is the in-memory cache ("store") of things that have been loaded. Removing the item from the list like this when it's deleted is the intuitive thing to do, and sometimes will work ok, but it does have a problem.

UI that needs the data to render will look in the store before making an API request. This happens, among other places, in `useJourneyInstance()` which uses `loadItemIfNecessary()`. If that function does not find the item in the cache, or for some other reason deems it necessary, it will request it from the API.

The problem with removing it like this is that `loadItemifNecessary()` will not find the cached item, so it will request it, which will fail (404) because it's now been deleted.

There's an alternative solution to do this, which is to set `deleted = true` on the item.

```suggestion
      state.journeyInstanceList.items = state.journeyInstanceList.items.map(
        (item) => item.id == id ? { ...item, deleted: true } : item
      );
```

That way, the item will still be returned by `useItemIfNecessary()` and not load again, but it will be excluded from `loadListIfNecessary()`.